File: fathersDay/Dsa/Day 21 (2697. Lexicographically Smallest Palindrome)/Solution.cpp
#include <iostream>
#include <string>
#include <algorithm>

class Solution {
public:
    std::string makeSmallestPalindrome(std::string s) {
        int left = 0;
        int right = s.length() - 1;
        while (left < right) {
            if (s[left] != s[right]) {
                if (s[left] < s[right]) {
                    s[right] = s[left];
                } else {
                    s[left] = s[right];
                }
            }
            left++;
            right--;
        }
        return s;
    }
};

int main() {
    Solution solution;
    std::string s = "egcfe";
    std::cout << solution.makeSmallestPalindrome(s) << std::endl; // Output: "efcfe"
    return 0;
}

File: fathersDay/Dsa/Day 21 (2697. Lexicographically Smallest Palindrome)/Solution.java
class Solution {
    public String makeSmallestPalindrome(String s) {
        char[] sArray = s.toCharArray();
        int left = 0;
        int right = sArray.length - 1;
        while (left < right) {
            if (sArray[left] != sArray[right]) {
                if (sArray[left] < sArray[right]) {
                    sArray[right] = sArray[left];
                } else {
                    sArray[left] = sArray[right];
                }
            }
            left++;
            right--;
        }
        return new String(sArray);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String s = "egcfe";
        System.out.println(solution.makeSmallestPalindrome(s)); // Output: "efcfe"
    }
}
